### PR TITLE
Internal: Create "Set up a header" step - PRO [ED-15629]

### PR DESCRIPTION
**Pull request by ronenboxer**

### CODE REVIEW

File: modules/checklist/steps/step-base.php [hunk 1]
```diff
@@ -173,9 +173,9 @@ public function get_is_locked() : bool {
 		return $this->is_locked;
 	}
 
-	public function set_is_locked( $is_locked ) : void {
-		$this->is_locked = $is_locked;
-	}
+//	public function set_is_locked( $is_locked ) : void {
```
Comment (TzviRabinovitch) [inline modules/checklist/steps/step-base.php:176]: delete

Comment (ronenelementor) [reply modules/checklist/steps/step-base.php]: Deleted

File: modules/checklist/steps-manager.php [hunk 2]
```diff
@@ -15,14 +15,17 @@
...
-	private array $steps_config = [];
 	private static array $step_ids = [ Add_Logo::STEP_ID, Create_Pages::STEP_ID, Setup_Header::STEP_ID, Assign_Homepage::STEP_ID ];
 
 	private Checklist_Module_Interface $module;
 
 	public function __construct( Checklist_Module_Interface $module ) {
 		$this->module = $module;
 		$this->register_steps();
+
+		add_action( 'elementor/init', function() {
```
Comment (louiswol94) [inline modules/checklist/steps-manager.php:26]: Why do we need to wrap this in `elementor/init`?

Comment (ronenelementor) [reply modules/checklist/steps-manager.php:26]: The `add_filter` fires after the `apply_filters` ran, so it didn't "listen" to it

Maybe another hook?

Comment (louiswol94) [reply modules/checklist/steps-manager.php:26]: @ronenelementor Maybe you need to use `do_action` to allow developers to register more steps.
Check Display Conditions:
`do_action( 'elementor/display_conditions/register', $this );`

Comment (louiswol94) [reply modules/checklist/steps-manager.php:26]: I created a meeting for us tomorrow morning if we need to talk about this further :)

**Review by davseve — DISMISSED** [review]

File: modules/checklist/assets/js/app/components/checklist-card-content.js [hunk 3]
```diff
@@ -13,18 +13,19 @@ const ChecklistCardContent = ( { step, setSteps } ) => {
...
 		image_src: imageSrc,
-		is_locked: isLocked,
-		promotion_url: promotionUrl,
+		promotion_data: promotionData,
 	} = step.config;
 
-	const ctaText = isLocked ? __( 'Upgrade Now', 'elementor-pro' ) : step.config.cta_text,
-		ctaUrl = isLocked ? promotionUrl : step.config.cta_url,
+	const ctaText = promotionData
+			? promotionData?.text || __( 'Upgrade Now', 'elementor-pro' )
```
Comment (louiswol94) [inline modules/checklist/assets/js/app/components/checklist-card-content.js:20]:
```suggestion
			? promotionData?.text || __( 'Upgrade Now', 'elementor' )
```

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**